### PR TITLE
[Feature] Unify load profile through analyze profile & support session/table profile collect granularity

### LINKED ISSUE #47069: Unify Load Profile
- [x] https://github.com/StarRocks/starrocks/pull/47548

### PR DESCRIPTION
**Pull request by meegoo**
## Why I'm doing:
* support table granularity for profile collect
* support profile collect when stream/routine load failure
* unify profile analyze interface

## What I'm doing:
```
mysql> create table t(k int, v int);
Query OK, 0 rows affected (0.11 sec)

mysql> alter table t set('enable_load_profile'='true');
Query OK, 0 rows affected (0.02 sec)

mysql> analyze profile from '1a462e3a-dcc7-94f4-e7ad-d272b485439f';
+-----------------------------------------------------------+
| Explain String                                            |
+-----------------------------------------------------------+
| Load:                                                     |
|   Summary:                                                |
|      - Query ID: 1a462e3a-dcc7-94f4-e7ad-d272b485439f     |
|      - Start Time: 2024-06-27 01:14:35                    |
|      - End Time: 2024-06-27 01:14:47                      |
|      - Query Type: Load                                   |
|      - Load Type: STREAM_LOAD                             |
|      - Query State: Finished                              |
|      - StarRocks Version: unify_load_profile-a79910b      |
|      - Sql Statement                                      |
|      - Default Db: d                                      |
|      - Total: 11s515ms                                    |
|   Execution:                                              |
|     Fragment 0:                                           |
|        - Address: 172.26.95.141:51686                     |
|        - InstanceId: 1a462e3a-dcc7-94f4-e7ad-d272b48543a0 |
|        - TxnID: 8009                                      |
|        - ReplicatedStorage: true                          |
|        - AutomaticPartition: false                        |
|        - AutomaticBucketSize: 4294967296                  |
|        - InstanceAllocatedMemoryUsage: 1.370 GB           |
|        - InstanceDeallocatedMemoryUsage: 860.124 MB       |
|        - InstancePeakMemoryUsage: 15.092 MB               |
|        - MemoryLimit: -1.000 B                            |
|        - RowsProduced: 55.325M (55324672)                 |
|          - AllocAutoIncrementTime: 1.286ms                |
|          - BytesRead: 0.000 B                             |
|          - CloseWaitTime: 0ns                             |
|          - IOTaskExecTime: 0ns                            |
|          - IOTaskWaitTime: 0ns                            |
|          - IndexNum: 1                                    |
|          - NumDiskAccess: 0                               |
|          - OpenTime: 2.942ms                              |
|          - PeakMemoryUsage: 0.000 B                       |
|          - PrepareDataTime: 1s386ms                       |
|            - ConvertChunkTime: 0ns                        |
|            - ValidateDataTime: 0ns                        |
|          - RowsFiltered: 0                                |
|          - RowsRead: 0                                    |
|          - RowsReturned: 55.325M (55324672)               |
|          - RowsReturnedRate: 69.136M (69135610) /sec      |
|          - RpcClientSideTime: 9s521ms                     |
|          - RpcServerSideTime: 0ns                         |
|          - RpcServerWaitFlushTime: 0ns                    |
|          - ScanTime: 8s815ms                              |
|          - ScannerQueueCounter: 199                       |
|          - ScannerQueueTime: 11s126ms                     |
|          - ScannerThreadsInvoluntaryContextSwitches: 0    |
|          - ScannerThreadsTotalWallClockTime: 0ns          |
|            - MaterializeTupleTime(*): 0ns                 |
|            - ScannerThreadsSysTime: 0ns                   |
|            - ScannerThreadsUserTime: 0ns                  |
|          - ScannerThreadsVoluntaryContextSwitches: 0      |
|          - SendDataTime: 8s728ms                          |
|            - PackChunkTime: 1s634ms                       |
|            - SendRpcTime: 0ns                             |
|              - CompressTime: 0ns                          |
|              - SerializeChunkTime: 0ns                    |
|            - WaitResponseTime: 2s238ms                    |
|          - TotalRawReadTime(*): 0ns                       |
|          - TotalReadThroughput: 0.000 B/sec               |
|       DataSource:                                         |
|          - DataSourceType: FileDataSource                 |
|          - FileScanner:                                   |
|            - CastChunkTime: 0ns                           |
|            - CreateChunkTime: 0ns                         |
|            - FileReadCount: 0                             |
|            - FileReadTime: 0ns                            |
|            - FillTime: 0ns                                |
|            - MaterializeTime: 0ns                         |
|            - ReadTime: 0ns                                |
|          - ScannerTotalTime: 0ns                          |
+-----------------------------------------------------------+
72 rows in set (0.02 sec)
```

Fixes #47069

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/administration/management/FE_configuration.md [hunk 1]
```diff
@@ -4524,9 +4524,9 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
 -->
 
 <!--
-##### stream_load_profile_collect_second
+##### load_profile_collect_threshold_second
```
Comment (wyb) [inline docs/en/administration/management/FE_configuration.md:4527]:
```suggestion
##### stream_load_profile_collect_threshold_second
```

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java [hunk 2]
```diff
@@ -872,7 +872,7 @@ public TExecPlanFragmentParams plan(TUniqueId loadId, long txnId, String label)
                     new StreamLoadPlanner(db, (OlapTable) table, info);
             TExecPlanFragmentParams planParams = planner.plan(loadId);
             planParams.query_options.setLoad_job_type(TLoadJobType.ROUTINE_LOAD);
-            if (planParams.query_options.enable_profile) {
+            if (true) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:875]: if true?

File: fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java [hunk 3]
```diff
@@ -911,11 +911,14 @@ public void updateFragmentExecStatus(TReportExecStatusParams params) {
             return;
         }
 
+        LOG.info("execState: {} params: {}", execState, params);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java:914]:
```suggestion
        LOG.debug("execState: {} params: {}", execState, params);
```

File: fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java [hunk 4]
```diff
@@ -911,11 +911,14 @@ public void updateFragmentExecStatus(TReportExecStatusParams params) {
...
+
         queryProfile.updateProfile(execState, params);
 
         lock();
         try {
             if (!execState.updateExecStatus(params)) {
+                LOG.info("execState: {} update status failed, status: {}", execState, params.status);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java:921]:
```suggestion
                LOG.debug("execState: {} update status failed, status: {}", execState, params.status);
```

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 5]
```diff
@@ -1310,7 +1310,7 @@ private long loadTxnBeginImpl(TLoadTxnBeginRequest request, String clientIp) thr
         // just use default value of session variable
         // as there is no connectContext for sync stream load
         ConnectContext connectContext = new ConnectContext();
-        if (connectContext.getSessionVariable().isEnableLoadProfile()) {
+        if (true) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:1313]: if true?

**Review by wyb — APPROVED** [review]

**Review by wangruin — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-3.3

**Pull request merged**